src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/PairReadWithRefBases.java

```diff
@@ -0,0 +1,51 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.reference.ReferenceBases;
+
+/**
+ * PairReadWithRefBases pairs the ReferenceBases overlapping the reads with each read and produces KV with read as
+ * the key.
+ *
+ * |---------- ref bases 1 ---------|      |-------------------- ref bases 2 --------------------|
+ * |-- read a --|   |---- read b ---|         |--------------- read c --------------|
+ * results in,
+ *
+ * |-- read a --|   |---- read b ---|         |--------------- read c --------------|
+ * |ref bases 1a|   |-ref bases 1b -|         |----------- ref bases 2c ------------|
+ *
+ * KV<read a, ref bases 1a>
+ * KV<read b, ref bases 1b>
+ * KV<read c, ref bases 2c>
+ * Precondition: each refbases must contain ALL of the bases for all of the GATKReads it's paired with.
+ */
+
+public class PairReadWithRefBases extends PTransform<PCollection<KV<ReferenceBases, Iterable<GATKRead>>>, PCollection<KV<GATKRead, ReferenceBases>>> {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public PCollection<KV<GATKRead, ReferenceBases>> apply(PCollection<KV<ReferenceBases, Iterable<GATKRead>>> input) {
+        return input.apply(ParDo.of(new DoFn<KV<ReferenceBases, Iterable<GATKRead>>, KV<GATKRead, ReferenceBases>>() {
+            private static final long serialVersionUID = 1L;
+            @Override
+            public void processElement(ProcessContext c) throws Exception {
```

Comment:
This sounds like a concern for the optimization phase of this project rather than the current "getting all the pieces in place and working phase" -- but I agree that being smart in our approach to the reference data will be critical.